src/app/(proper_react)/(onboarding)/user/welcome/EnterInfo.tsx

```diff
@@ -4,22 +4,357 @@
 
 "use client";
 
+import Image from "next/image";
 import { Session } from "next-auth";
-import styles from "./EnterInfo.module.scss";
+import { useEffect, useRef, useState } from "react";
+import { useOverlayTriggerState } from "react-stately";
+import {
+  useOverlayTrigger,
+  AriaTextFieldProps,
+  useTextField,
+} from "react-aria";
+import whyWeNeedInfoHero from "./images/welcome-why-we-need-info.svg";
 import { useL10n } from "../../../../hooks/l10n";
+import { ModalOverlay } from "../../../../components/client/dialog/ModalOverlay";
+import { Dialog } from "../../../../components/client/dialog/Dialog";
+import { Button } from "../../../../components/server/Button";
+
+import enterInfoStyles from "./EnterInfo.module.scss";
+import viewStyles from "./View.module.scss";
 
 export type Props = {
   onDataSaved: () => void;
+  onGoBack: () => void;
   user: Session["user"];
 };
 
-export const EnterInfo = (_props: Props) => {
-  const _l10n = useL10n();
+// TODO: Move into separate component and add to storybook
+function TextField(props: AriaTextFieldProps) {
+  const inputRef = useRef(null);
+  const { label, value } = props;
+  const { errorMessageProps, inputProps, labelProps } = useTextField(
+    props,
+    inputRef
+  );
+  const { errorMessage, validationState } = props;
+  const showError = errorMessage && validationState === "invalid";
+
+  return (
+    <div className={enterInfoStyles.input}>
+      <label {...labelProps} className={enterInfoStyles.inputLabel}>
+        {label}*
+      </label>
+      <input
+        {...inputProps}
+        ref={inputRef}
+        className={`${enterInfoStyles.inputField} ${
+          !value ? enterInfoStyles.noValue : ""
+        } ${showError ? enterInfoStyles.hasError : ""}`}
+      />
+      {showError && (
+        <div {...errorMessageProps} className={enterInfoStyles.inputMessage}>
+          {errorMessage}
+        </div>
+      )}
+    </div>
+  );
+}
+
+// TODO: Add more sophisticated for location data
+const getIsValidInfo = (value: string) => value !== "";
```

Comment:
Additional backend validation will be handled after https://github.com/mozilla/blurts-server/pull/3157 landed.